src/gleam/string.gleam

```diff
@@ -739,3 +739,70 @@ pub fn to_option(s: String) -> Option(String) {
     _ -> Some(s)
   }
 }
+
+/// Returns the first element in a grapheme and wraps it in an `Option(String)`.
+/// If the `String` is empty, it returns `None`. Otherwise, it returns
+/// `Some(String)`.
+///
+/// ## Examples
+///
+/// ```gleam
+/// > first("")
+/// None
+/// ```
+///
+/// ```gleam
+/// > first("icecream")
+/// Some("i")
+/// ```
+///
+pub fn first(s: String) -> Option(String) {
```

Comment:
The standard library convention is to return a result so that it can be used with try. Let's do that rather than option